dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/CrudControllerAdvice.java

```diff
@@ -136,16 +135,11 @@ protected void initBinder( WebDataBinder binder )
         this.enumClasses.forEach( c -> binder.registerCustomEditor( c, new ConvertEnum( c ) ) );
     }
 
-    @ExceptionHandler( org.hisp.dhis.feedback.BadRequestException.class )
```

Comment:
I don't think we should do that. AFAIK BadRequestException/ForbiddenException were recently introduced by @jbee. Is there any issue with us simply using BadRequestException/ForbiddenException from `org.hisp.dhis.feedback`?

We want to return HTTP status 400/403 by throwing an exception in our controllers. If I understand correctly it would also have the advantage of showing up in the OpenAPI spec.

Reply:
You are right, for some reason I thought I created that method myself in a previous commit, I just rolled it back to the way it was. 

The thing about using `org.hisp.dhis.feedback.BadRequestException` is that we have to add the exception to the method signature, whilst using org.hisp.dhis.common.BadRequestException we don't. Since it's used in a lot of places, I thought we'd be better off that way, but if we favor OpenAPI, I'll change it.

Reply:
AFAIK making it a checked exception was intentional as it allows us to extract that information and put it into the OpenAPI spec. Right @jbee ?

Reply:
Making these checked was intentional. By forcing them to be listed in the `throws` they propagate to all the places that somehow might call that code. This makes sure all exceptions that go along with a specific HTTP status code are aggregated up to the controller endpoint method. 

By making them checked you cannot miss an intentional error status code an endpoint might have as the compiler forces you to either propagate or handle. This helps OpenAPI but I'd argue it also helps the programmer to understand directly from the method signatures what error they might cause, or if you don't want this to be exposed in the REST API that a try-catch at some level would be required. 

It also helps to stay true to a contract that previously was only encoded implicitly or by convention. For example the pre- and post- hook handlers should never intentionally throw something that not ends up as a `Conflict` status. Now this contract is clearly stated by those methods only throwing a `ConflictException`.  

Having said all that this does not mean you should never use unchecked exceptions. It just means failures that you want to propagate to the REST API should use these checked exceptions as they are part of the API. This includes the code that throws them. Classic example would be validation.